subsys/usb/class/hid/core.c

```diff
@@ -123,11 +143,54 @@ static int hid_on_get_protocol(struct usb_setup_packet *setup, s32_t *len,
 static int hid_on_set_idle(struct usb_setup_packet *setup, s32_t *len,
 			   u8_t **data)
 {
-	LOG_DBG("Set Idle callback");
+#ifdef CONFIG_USB_DEVICE_SOF
+	u8_t rate = ((sys_le16_to_cpu(setup->wValue) & 0xFF00) >> 8);
+	u8_t report_id = sys_le16_to_cpu(setup->wValue) & 0xFF;
 
-	/* TODO: Do something. */
+	if (report_id > CONFIG_USB_HID_REPORTS) {
+		LOG_ERR("Report id out of limit: %d", report_id);
+		return -ENOTSUP;
+	}
+
+	LOG_DBG("Set Idle callback, rate: %d, report_id: %d", rate, report_id);
+
+	hid_device.idle_rate[report_id] = rate;
 
+	if (rate == 0) {
+		/* Clear idle */
+		bool clear = true;
+		u8_t i = 0;
+
+		for (i = 1; i <= CONFIG_USB_HID_REPORTS; i++) {
```

Comment:
I think you can declare loop variable here (or is there a new rule against it?). The same on other places below.

Reply:
Done.